synapse/config/emailconfig.py

```diff
@@ -74,19 +74,33 @@ def read_config(self, config, **kwargs):
             "renew_at"
         )
 
-        email_trust_identity_server_for_password_resets = email_config.get(
-            "trust_identity_server_for_password_resets", False
+        self.email_threepid_behaviour = (
+            # Have Synapse handle the email sending if account_threepid_delegate
+            # is not defined
+            "remote"
+            if self.account_threepid_delegate
+            else "local"
         )
-        self.email_password_reset_behaviour = (
-            "remote" if email_trust_identity_server_for_password_resets else "local"
-        )
-        self.password_resets_were_disabled_due_to_email_config = False
-        if self.email_password_reset_behaviour == "local" and email_config == {}:
+        # Prior to Synapse v1.4.0, there used to be another option that defined whether Synapse
+        # would use an identity server to password reset tokens on its behalf. We now warn the
+        # user if they have this set and tell them to use the updated option.
+        # TODO: Eventually we want to remove the functionality of having an identity server
+        #  send tokens on behalf of the homeserver. At that point, we should remove this check
```

Comment:
I'm not sure this is a helpful `TODO` TBH, and if we want to keep it somewhere then let's make an issue.

Reply:
https://github.com/matrix-org/synapse/issues/5881